ITE/4/README.adoc

```diff
@@ -0,0 +1,462 @@
+(NTIA) link:https://www.ntia.doc.gov/SoftwareTransparency[Software Component Transparency].
+SPDX and in-toto, along with representatives from NTIA and other stakeholders,
+are also part of the Consortium for Information and Software Quality (CISQ) and
+Object Management Group's (OMG) working group on
+link:https://www.it-cisq.org/software-bill-of-materials/index.htm[Tool-to-tool Software Bill of Materials Exchange].
+
+SPDX documents composed of several entities that have unique identifiers. These
+identifiers can be used in in-toto metadata embedded in SPDX documents to refer
+to the respective entities.
+
+===== in-toto link attestation for packaging SPDX files into an SPDX package
+
+An in-toto attestation can be bundled into an SPDX document to show the chain of
+custody for the elements the SPDX document refers to. Here, we see how
+provenance can be attested to for File and Package entities in an SPDX document.
+
+```
+{
+    "_type": "link",
+    "name": "package-ghostscript-9.21.tar.gz",
+    "command": "<COMMAND>",
+    "materials": {
+        "spdx:SPDXRef-141-File-83pv-RKSJ-H-d51620a4d7d9aeca3a1cbe5ef201513f98d65f98": <HASH>,
+        "spdx:SPDXRef-271-File-AUTHORS.md-109c93392646b4d55e3ca62c5b578a9ac7cc159f": <HASH>,
+        "...": "..."
+    },
+    "products": {
+        "spdx:SPDXRef-Pkg-ghostscript-9.21.tar.gz-6f60d7fcb5eef6a8bec5abedf21c6a7008a8c0c7": <HASH>
+    },
+    "byproducts": {
+        "stderr": "",
+        "stdout": "",
+        "return-value": ""
+    },
+    "environment": {
+        "variables": "",
+        "filesystem": "",
+        "workdir": ""
+    }
+}
+```
+
+==== Use Case 2: GitHub Entities
+
+GitHub has more abstract entities such as Pull Requests and Issues. These
+entities can be referred to directly using the URI schemes proposed in this ITE
+and help provide attestations about these artifacts. Consider:
+
+===== in-toto link attestation for creating a pull request
+
+A pull request is a proposal to make changes to a repository. Changes are either
+made on a separate branch on the same repository or a branch on a fork of the
+repository and the pull request is a proposal to merge these changes into the
+main repository.
+
+```
+{
+    "_type": "link",
+    "name": "pull-request-250",
+    "command": "",
+    "materials": {
+        "github+https://github.com/in-toto/in-toto/commit/3371c93699785ba5907411a321ce82c59cb127fa": <HASH>,
+        "...": "..."
+    },
+    "products": {
+        "github+https://github.com/in-toto/in-toto/pull/250": <HASH>
+    },
+    "byproducts": {
+        "stderr": "",
+        "stdout": "",
+        "return-value": ""
+    },
+    "environment": {
+        "variables": "",
+        "filesystem": "",
+        "workdir": ""
+    }
+}
+```
+
+The materials aren't limited to the commits that make up a pull request, but can
+also contain other elements such as reviews or other comments that make up a
```

Comment:
If it contains reviews and/or comments which may grow over time, how can you compute a hash over it?

Reply:
As it stands, an in-toto link records the hashes of artifacts at the time of generation of the link. This doesn't change with this ITE. A link generated for a GitHub PR would capture the information in the PR at that time. If I'm understanding it right, the question then becomes "when is the link generated for a PR", and I suspect that becomes an implementation detail that shouldn't be resolved in this ITE.